packages/nimble-components/src/table/tests/table.stories.ts

```diff
@@ -363,6 +382,11 @@ const metadata: Meta<TableArgs> = {
                 'A function that returns `true` if the configuration of the table is valid and `false` if the configuration of the table is not valid.',
             control: false
         },
+        addDelayedChildrenIfNeeded: {
+            name: 'setRecordHierarchyOptions()',
+            description: setRecordHierarchyOptionsDescription,
+            control: false
```

Comment:
nit: I have a mild preference to show some kind of control for any paramaters that are writable. It's less helpful here since the configuration won't impact what shows in the code preview, but it can still be nice to be able to play with a setting toggled on and off.

The simplest configuration I can think of would be two radio buttons with names something like "No delayed children" and "Last record has delayed children" but I'm open to other ideas.

I'm also open to deferring this until the progress indicator PR or scrapping the idea if it complicates the implementation too much.

Reply:
What about making it a separate story? like how the table column configuration it broken into separate workflow:
![image](https://github.com/ni/nimble/assets/1588923/428799aa-29e3-4b4e-bfa4-de6967a09cec)

Can have a "Table/Delayed Loading" story and it can have a control where you explicit set the state of a row with different options:
- default
- hasChildren
- loading

And can move the expected workflow discussion into Usage Docs for that story

Reply:
I think a separate story within the same MDX could work. Another way to split it would be to put all the flat data examples in one story and the hierarchical data examples (both pre-loaded and delay-loaded) in a second story. No strong preference from me, whatever works better as you start moving parameters around.

Brainstorming on that idea a little further, maybe we could split up the huge parameter table we currently have into a few smaller ones that each live in their own section within the MDX. e.g. Maybe Validity be its own section that's just a parameter table, no embedded nimble-table. Just brainstorming, that doesn't have to be in this PR.

Reply:
I've updated the stories to split out delayed-hierarchy into its own story, with more of the documentation in the mdx file rather than within the controls table.

Reply:
I like how it looks! But in [the current build](https://60e89457a987cf003efc0a5b-owxluyyebl.chromatic.com/?path=/docs/components-table--docs) clicking the expander for the delay loaded rows doesn't ever populate the rows with data. (macOS Firefox in case that matters)

It was also slightly unexpected to see the "Localizable labels" entry in the parameter table for delay-loaded hierarchy.

https://github.com/ni/nimble/assets/10500124/8a5df9a6-9458-49c1-a38a-17070f7114b5

Reply:
> clicking the expander for the delay loaded rows doesn't ever populate the rows with data

That is correct. The row state is now completely driven by the story's control. It seemed odd to have the state coming from the control but then change the state out from under the user when they interact with the table, but maybe that's better than doing nothing when the expand button is clicked?

Reply:
Ah, gotcha, I was still stuck on the previous behavior. I can see both sides. It'll be useful to have demos that clearly show the effect of putting a row into each state. And it'll be useful to have a demo that automatically transitions a row through the states to show how the feature behaves as a whole (and what a client has to do to achieve that).

I'm not sure of the best way to show both of those demos and I'm about to be OOO for a while so I'll have to leave you to figure it out. But I'm fine with keeping it as you have it for this PR and then adding the more comprehensive demo (either in storybook or the example app) in a follow up once the loading progress indicator is visible.

Reply:
Maybe instead of the exact row state the control can be phrased in the workflow steps and describe the expectations better. If we experiment with that lets do it in a separate PR, keeping the extra complexity of this niche / not common workflow out of the main table story was my goal.

Reply:
I'm leaving this as-is for now, but I'm happy to iterate on how we present this workflow in storybook in a separate PR.